Attack.py

```
@@ -102,1366 +86,676 @@ def __call__(self, *args: Any, **kwargs: Any) -> "_NXStub":
+    if a == 0:
+        return b, 0, 1
+    g, x1, y1 = extended_gcd(b % a, a)
+    return g, y1 - (b // a) * x1, x1
+
+def shannon_entropy(arr: np.ndarray) -> float:
+    """Shannon entropy in bits (object-dtype safe)."""
+    if arr.size == 0:
+        return 0.0
+    if SCIPY_AVAILABLE:
+        return float(scipy_entropy(arr, base=2))
+    uniq, counts = np.unique(arr, return_counts=True)
```

Ruff (0.14.2)

Unpacked variable `uniq` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)